### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by sarthak77**
Creating a readme for alerting service for RazorPay folks for a better understanding of alerting service offered by hypertrace.
Reference Issue: https://github.com/hypertrace/hypertrace-alert-engine/issues/40

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,95 @@
...
+
+  2. violationCondition/baseline
+
+      Baseline can be fixed or dynamic (based on past data points). Currently we support fixed baseline based on which anomaly would be decided. As an example the above rule uses the baseline : "Raise an anomaly when the selected metric value is greater than 5 for all the points in the 15s duration"
+
+### Metric Anomaly Evaluation Engine
+At a time there can be multiple rules configured. So we need a mechanism to evaluate all the rules and at all the times for alert to be useful. We have an evaluation engine which in background, keeps on evaluating all the rules in a periodic fashion at every 1 minute interval. At a high level, for a given rule the evaluation will look like as:
+
+1. Fetch all the metric points of metric as per selection definition for current 1 min duration by firing query to the data store (pinot in our case). So, if granularity is 15s, there will be a maximum of 4 points will be received, and if it's 1 min, there will be 1 data point.
```
Comment (rish691) [inline README.md:84]: Compress these details into a single paragraph

File: README.md [hunk 2]
```diff
@@ -1 +1,95 @@
...
+tasks. More features would be added as we progress along in time.
+
+- Get an alert whenever there is a sudden spike/ sharp increase in latency of any operation/ API calls from or to my service.
+- Get an alert whenever there is a sudden spike in traffic/ increase in call rate to my service/ API. 
+- Get an alert whenever there is a sudden spike in errors/ increase in error rate for my service. 
+
+## Design
+Currently, we support setting up the rules via a config or via deployment. However, in the UI, we will have a way to list the alert rules and notification configuration. If we closely look at the alerting service, we see that it is simply an anomaly condition from a baseline for a particular metric. So at a high level we need a **metric anomaly definition** (alert rule) which consists of metric selection and baseline calculation. Once we have designed our alert rule, we would need a **anomaly evaluation engine** over a metric (alert rule evaluation engine) and an **alert notification engine** (notification sender) to send the notification.
```
Comment (rish691) [inline README.md:12]: > So at a high level we need a

this is referring to component which already exists, please appropriately fix the language used

File: README.md [hunk 3]
```diff
@@ -1 +1,91 @@
-# alerting
\ No newline at end of file
+# Alerting
+
+## Introduction
+With Hypertrace we are moving in a direction where we promote it more like an Observability platform than just a distributed tracing tool similar to Jaeger or Zipkin. So when we talk about observability, alerting is one of the main components of it. As of now we provide support for the following 
+tasks. More features would be added as we progress along in time.
```
Comment (JBAhire) [inline README.md:5]:
```suggestion
Hypertrace is the modern Observability platform and when we talk about Observability, these are the four pillars of the Observability Engineering team’s charter:
- Monitoring
- Alerting/visualization
- Distributed systems tracing infrastructure
- Log aggregation/analytics

Alerting is the entry point for most of the debugging workflows and all the modern DevOps teams are required to have a reactive mechanism in a place to quickly identify and mitigate the failures. 
```

File: README.md [hunk 4]
```diff
@@ -1 +1,91 @@
...
+
+- Get an alert whenever there is a sudden spike/ sharp increase in latency of any operation/ API calls from or to my service.
+- Get an alert whenever there is a sudden spike in traffic/ increase in call rate to my service/ API. 
+- Get an alert whenever there is a sudden spike in errors/ increase in error rate for my service. 
+
+## Design
```
Comment (JBAhire) [inline README.md:11]: Add a high level architecture diagram with alerting component if possible.

File: README.md [hunk 5]
```diff
@@ -1 +1,91 @@
...
+
+### Metric Anomaly Evaluation Engine
+At a time there can be multiple rules configured so we need a mechanism to evaluate all the rules and at all the times for alert to be useful/appropriate. For this we have an evaluation engine which in background, keeps on evaluating all the rules in a periodic fashion at every configured time interval (say every 1 minute). At a high level, for a given rule the evaluation engine will first fetch all the metric points of metric as per selection definition for configured time by firing query to the data store (pinot in our case). For example say if configured time is 1 minute, and granularity is 15s, there will be a maximum of 4 points received, and if it's 1 min, there will be 1 data point received. Similarly, the engine will fetch all the metrics points for the required baseline for all the rules and then evaluate all these points against the violation condition. For one alert rule,if all points violate, alert is raised and send via slack/email.
+
+
+### Alert Notification Engine
```
Comment (JBAhire) [inline README.md:85]: Add screenshot of alert.

**Review by JBAhire — APPROVED** [review]

**Comment by JBAhire** [comment]
Thanks, @sarthak77! Please make the changes I suggested and then we can merge it.

File: README.md (hunk 4 above)
Comment (sarthak77) [reply README.md:11]: Arch diagram on this [page](https://traceableai.atlassian.net/wiki/spaces/Engineering/pages/1443397832/Alerting+details#Arch-diagram) ?

**Comment by sarthak77** [comment]
Images not rendering on preview.

File: README.md [hunk 6]
```diff
@@ -1 +1,98 @@
...
+
+- Get an alert whenever there is a sudden spike/ sharp increase in latency of any operation/ API calls from or to my service.
+- Get an alert whenever there is a sudden spike in traffic/ increase in call rate to my service/ API. 
+- Get an alert whenever there is a sudden spike in errors/ increase in error rate for my service. 
+
+## Design
+Currently, we support setting up the rules via a config or via deployment. However, in the UI, we will have a way to list the alert rules and notification configuration. If we closely look at the alerting service, we see that it is simply an anomaly condition from a baseline for a particular metric. So to meet the requirements, we have a **metric anomaly definition** (alert rule) which consists of metric selection and baseline calculation steps. Then we have an **anomaly evaluation engine** which evaluates this rule over a metric (alert rule evaluation engine) and an **alert notification engine** (notification sender) to send the notifications for alerting. Overall architecture diagram now looks like:
+![hypertrace architecture diagram](https://ibb.co/rQvVJDP)
```
Comment (JBAhire) [inline README.md:18]: you have to add a link to image file i.e. a link to png/ jpeg.

**Review by rish691 — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -1 +1,101 @@
...
+
+- Get an alert whenever there is a sudden spike/ sharp increase in latency of any operation/ API calls from or to my service.
+- Get an alert whenever there is a sudden spike in traffic/ increase in call rate to my service/ API. 
+- Get an alert whenever there is a sudden spike in errors/ increase in error rate for my service. 
+
+## Design
+Currently, we support setting up the rules via a config or via deployment. However, in the UI, we will have a way to list the alert rules and notification configuration. If we closely look at the alerting service, we see that it is simply an anomaly condition from a baseline for a particular metric. So to meet the requirements, we have a **metric anomaly definition** (alert rule) which consists of metric selection and baseline calculation steps. Then we have an **anomaly evaluation engine** which evaluates this rule over a metric (alert rule evaluation engine) and an **alert notification engine** (notification sender) to send the notifications for alerting. Overall architecture diagram now looks like:
+
+![ht-architecture](https://github.com/hypertrace/hypertrace-alert-engine/blob/readme/images/ht-architecture.png)
```
Comment (JBAhire) [inline README.md:19]:
```suggestion
![ht-architecture](../images/ht-architecture.png)
```

Comment (JBAhire) [reply README.md:19]: don't use link because this `readme` branch will be deleted once you merge PR and it will fail.

**Pull request merged**